src/backend/gpopt/gpdbwrappers.cpp

```diff
@@ -2735,4 +2747,35 @@ gpdb::IsTypeRange(Oid typid)
 	return false;
 }
 
+bool
+gpdb::MarkMDCacheAsTransient(Relation index_rel)
+{
+	GP_WRAP_START;
+	{
+		bool result =
+			index_rel->rd_index->indcheckxmin &&
+			!TransactionIdPrecedes(
+				HeapTupleHeaderGetXmin(index_rel->rd_indextuple->t_data),
+				TransactionXmin);
+		if (result)
+			mdcache_transaction_xmin = TransactionXmin;
+		return result;
+	}
+	GP_WRAP_END;
+	// ignore index if we can't check it visibility for some reason
+	return true;
+}
+
+void
+gpdb::UnMarkMDCacheAsTransient()
+{
+	mdcache_transaction_xmin = InvalidTransactionId;
+}
+
+bool
+gpdb::IsMDCacheTransient()
+{
+	return TransactionIdIsValid(mdcache_transaction_xmin);
```

Comment:
Why did you remove the wrappers here?

Reply:
this macro is revealed in just a comparison.